kokoro/gcp_ubuntu/releases/kokoro_build_release.sh

```diff
@@ -0,0 +1,130 @@
+# limitations under the License
+#
+# A Kokoro build script to build release and update github repo
+
+# Code under repo is checked out to ${KOKORO_ARTIFACTS_DIR}/github.
+# The final directory name in this path is determined by the scm name specified
+# in the job configuration.
+
+# Display commands being run.
+# WARNING: please only enable 'set -x' if necessary for debugging, and be very
+#  careful if you handle credentials (e.g. from Keystore) with 'set -x':
+#  statements like "export VAR=$(cat /tmp/keystore/credentials)" will result in
+#  the credentials being printed in build logs.
+#  Additionally, recursive invocation with credentials as command-line
+#  parameters, will print the full command, with credentials, in the build logs.
+# set -x
+# Fail on any error.
+set -e
+
+#Variables
+export GIT_PSW="$(<${KOKORO_KEYSTORE_DIR}/76474_bqassessment-github-releases-02032022)"
+export GIT_RELEASES_USERNAME="edw-assessment-integration-testing-bot"
+export KOKORO_BUILD_RELEASE_DIR="${KOKORO_ARTIFACTS_DIR}/github/dwh-assessment-extraction-tool"
+export KOKORO_RELEASE_OUTPUT_FILE="${KOKORO_BUILD_RELEASE_DIR}/bazel-bin/dist/dwh-assessment-extraction-tool.zip"
+export SCRIPT_DIR=$(dirname "$0")
+export SCRIPT_PARENT_DIR="$(dirname -- "$SCRIPT_DIR")"
+export BUILD_SCRIPT="$SCRIPT_PARENT_DIR/kokoro_build.sh"
+source ${SCRIPT_DIR}/release_utils.sh
+
+cd ${KOKORO_BUILD_RELEASE_DIR}
+
+log "Current dir "$(pwd)
+git checkout main
+
+export LAST_GIT_TAG=$(git tag  \
+    | grep -E '^v[0-9]' \
+    | sort -V \
+    | tail -1 )
+
+# Do we already know what version we want to release?
+if [[ -z "${CREATE_TAG}" ]]; then
+  if [[ -z "${LAST_GIT_TAG}" ]]; then
+    err "No previous git tag found and it was not provided with CREATE_TAG env"
+  fi
+  VERSION=$(incrementTagVersion ${LAST_GIT_TAG}) 
+else
+  VERSION="${CREATE_TAG}"
+fi
+
+log "Will create new version "${VERSION}
+
+if [ $(git tag -l "$VERSION") ]; then
+  err "ERROR! Tag for specified version already exists! ${VERSION}"  # write error message to stderr, exits
+fi
+
+code=$(httpGetErrorCode "Accept: application/vnd.github.v3+json" \
+  "https://${GIT_RELEASES_USERNAME}:${GIT_PSW}@api.github.com/repos/google/dwh-assessment-extraction-tool/releases/tags/${VERSION}" )
+if [ $code != "404" ]; then
+  err "ERROR! Release with ${VERSION} tag version name already exists or http failed! Http code is ${code}" 
+fi
+
+log "New version name verified"
+
+#Run build and integration tests
+log "Build script is to be invoked here: "$BUILD_SCRIPT
+
+# TODO : would need to be reenabled once in main
+#sh $BUILD_SCRIPT   
+
+# TODO: temp taken from kokoro_build.sh, need to be removed later
+use_bazel.sh 4.1.0
+command -v bazel
+bazel version
+cd "${KOKORO_ARTIFACTS_DIR}/github/dwh-assessment-extraction-tool"
+#Build extraction tool
+bazel build dist:all
+# TODO: temp taken from kokoro_build.sh END
+
+# revert to initial state after running build script
+cd "${KOKORO_BUILD_RELEASE_DIR}"
+git checkout main
```

Comment:
Do not fully understand this line, we already on main (line 46) at this point. Checking out the same branch doesn't lead to reverting to the initial state.

Maybe there is something in the BUILD_SCRIPT which makes this valid?

Reply:
Indeed. In theory there shouldn't be any need for that at the same time I am calling a build script out of my contrrol so I want to make sure with state I can expect. Once they have their script in main I will remove this line

Reply:
Should we not rather trust that Kokoro gives us the right commit? I understand line 46 forces getting main because you are currently in dev but why check out main twice?

An alternative might be to let the user tag the release candidate beforehand and then check out the tag.